Support displaying various currencies instead of just XRP

sharon-wang opened the pull request:
Fixes: #97 

To keep things simple for our initial hackathon submission, we decided to only worry about displaying sent assets in XRP, as it was generally the default currency being used. However, on some sites (eg. YouTube), the assetCode used for streamed payment may not be XRP.

To support various currencies, I have made a few changes to the UI:
- top site circles no longer display amount of payment streamed in the circle, since multiple currencies could be at play and that would be too much text
    - I can change this to continue to display the amount of payment streamed in the circle ,if only one currency has ever been streamed to the origin, if that is preferred
- top site detail may contain a list of payments in different currencies that have been sent to the origin
- the main akita view will list various currencies streamed, if that has occurred

I have replaced our XRP-specific calculations with summing up the total payments being made in each currency for each origin.

Some screenshots to show the UI changes:
![image](https://user-images.githubusercontent.com/25834218/103338919-2fc78f00-4a3d-11eb-9c01-ade82a3052f1.png)
![image](https://user-images.githubusercontent.com/25834218/103338932-3ce47e00-4a3d-11eb-8400-b5be448aa5e3.png)

sharon-wang commented:
I also changed the decimal precision to be 2 decimals for simplicity. Doesn't work the best for crypto (since we're likely dealing with pretty small amounts), but is reasonable for XRP.

vezwork commented:
Just a tangent idea: in the top site detail it says (if you haven't sent payment to that site)

> If you were using Coil you would have sent ____ to this site.

Maybe this would be a bit more explicit why you're seeing that text:

> You haven't sent anything to this site. In the time you've spent here, Coil would have sent ____.

sharon-wang commented:
Yes I can update the estimation text with some better wording.

What do you think about moving `getTotalSentAssets(originStats)` and `getTotalSentAssetsForOrigin(originData)` from `src/main.js` to their respective classes `AkitaOriginStats` and `AkitaOriginData`? Maybe it'd make more sense as a class function instead of a general function in `main.js`.

vezwork commented:
Seeing this:
<img src="https://user-images.githubusercontent.com/11082236/103383398-d51b4b00-4aaf-11eb-86ae-abdadebfbf8c.png" height=400>
Once I spent a bit of time here it started to show that I have 0.01XRP

Edit: discussed with @sharon-wang and this is intended behavior.

vezwork commented on src/main.js:
```diff
@@ -153,44 +153,102 @@ function getEstimatedPaymentForTimeInUSD(timeSpent) {
-async function calculateTotalSentXRP() {
-	const originDataList = await getOriginDataList();
-	let totalSentXRP = 0;
+function getTotalSentAssets(originStats) {
```
There's no js-doc for this function, please add some.

vezwork commented on src/main.js:
```diff
@@ -153,44 +153,102 @@ function getEstimatedPaymentForTimeInUSD(timeSpent) {
-				totalSentXRP += actualAmount;
+			for (const sentAssetData of Object.values(paymentPointerData.sentAssetsMap)) {
+				const currency = sentAssetData.assetCode;
+				let amount = sentAssetData.amount * (10**(-sentAssetData.assetScale));
```
nit: pull this conversion out to a function.

vezwork commented on src/main.js:
```diff
@@ -153,44 +153,102 @@ function getEstimatedPaymentForTimeInUSD(timeSpent) {
+			for (const sentAssetData of Object.values(paymentPointerData.sentAssetsMap)) {
+				const currency = sentAssetData.assetCode;
+				let amount = sentAssetData.amount * (10**(-sentAssetData.assetScale));
+				const existingAmountForCurrency = sentAssetsMap.get(currency);
```
nit: move this inside the following `if` statement since it's only used inside there.

vezwork reviewed (changes requested):
Changes look great, just need js-doc on `getTotalSentAssets()`

vezwork commented:
> What do you think about moving `getTotalSentAssets(originStats)` and `getTotalSentAssetsForOrigin(originData)` from `src/main.js` to their respective classes `AkitaOriginStats` and `AkitaOriginData`? Maybe it'd make more sense as a class function instead of a general function in `main.js`.

Yea that makes more sense, do it!

sharon-wang commented:
Added missing jsdoc, moved the mentioned functions into the classes and moved the amount calculation into WebMonetizationAsset.

Updated wording, open to further edits though:

![image](https://user-images.githubusercontent.com/25834218/103386848-00a53200-4abe-11eb-93de-a985426564a2.png)

vezwork commented:
I like the wording, I like the changes, lgtm!

sharon-wang commented:
An example on how to test if the UI is displaying multiple currencies properly:

Add the following to the `main()` function in `src/content_scripts/content_origin.js` and save the file:
```
	await storeDataIntoAkitaFormat({
		paymentPointer: "testPaymentPointer",
		assetCode: "USD",
		assetScale: 9,
		amount: 1234567890
	}, AKITA_DATA_TYPE.PAYMENT);
	await storeDataIntoAkitaFormat({
		paymentPointer: "testPaymentPointer",
		assetCode: "CAD",
		assetScale: 9,
		amount: 1234567890
	}, AKITA_DATA_TYPE.PAYMENT);
	await storeDataIntoAkitaFormat({
		paymentPointer: "testPaymentPointer",
		assetCode: "XRP",
		assetScale: 9,
		amount: 1234567890
	}, AKITA_DATA_TYPE.PAYMENT);
```

1. Install Akita from scratch and have Coil enabled
2. Visit any site
3. Open Akita: the three currencies and the amounts we added should display
